From 9d3a0212621e6c2f41c19223843f66473299f3eb Mon Sep 17 00:00:00 2001
From: Quentin Young <qlyoung@nvidia.com>
Date: Tue, 29 Mar 2022 16:16:35 -0400
Subject: [PATCH] tests: skip grpc test if frr not installed

it wants yang models installed which will only be there if frr has been
installed before, causing `make check` to fail when run on a system on
which frr has not been installed when GRPC is enabled (--enable-grpc)

Signed-off-by: Quentin Young <qlyoung@nvidia.com>
---
 tests/lib/test_grpc.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/lib/test_grpc.py b/tests/lib/test_grpc.py
index 2e292fadc99..7f722de4222 100644
--- a/tests/lib/test_grpc.py
+++ b/tests/lib/test_grpc.py
@@ -13,6 +13,10 @@ class TestGRPC(object):
         'S["GRPC_TRUE"]=""\n' not in open("../config.status").readlines(),
         reason="GRPC not enabled",
     )
+    @pytest.mark.skipif(
+        not os.path.isdir("/usr/share/yang"),
+        reason="YANG models aren't installed in /usr/share/yang",
+    )
     def test_exits_cleanly(self):
         basedir = os.path.dirname(inspect.getsourcefile(type(self)))
         program = os.path.join(basedir, self.program)
